### PR TITLE
Add initial documentation for creating your own API

### PR DESCRIPTION
**Pull request by Zeegaan**
## Description

Added documentation about getting started with management API

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [x] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)
V14

## Deadline (if relevant)

 V14 Release 🚀

### CODE REVIEW

**Comment by eshanrnh** [comment]
Thanks for the PR, @Zeegaan 🙌We will look at it as soon as we can 💪

File: 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+[VersionedApiBackOfficeRoute("my/item")]
+[ApiExplorerSettings(GroupName = "My item API")]
+public class MyItemApiController : ManagementApiControllerBase
+{
+}
+```
+
+## Retrieving all items
+
+Now that we have our class set up, we can add an action to get all items. We use the `HttpGet` attribute to define the HTTP method and route for the action.
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md:19]:
```suggestion
Now that we have our class set up, we can add an action to get all items. We will use the `HttpGet` attribute to define the HTTP method and route for the action.
```

File: 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+The AllItems field is an in memory list of items to simulate use of a repository.
+We use `skip` & `take` parameters here, so users of this endpoint can implement paging.
+We also use the `PagedViewModel` to return the given items (10 by default), and then the total number of items.
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md:22]:
```suggestion
The `AllItems` field is an in memory list of items to simulate the use of a repository.
We use the `skip` & `take` parameters here, so users of this endpoint can implement paging.
We also use the `PagedViewModel` to return the given items (10 by default), and then the total number of items.
```

File: 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+    NotFound,
+    InvalidValue,
+    DuplicateValue
+}
+```
+## Creating a new item
+
+Now we can add an action to create a new item. We use the `HttpPost` attribute to define the HTTP method and route for the action.
+Here we can see some validation logic. If the value does not start with "New", we return a BadRequest response with an error message. This highlights why we use the `OperationStatusResult` method. We can return a detailed response.
+Here we also use `CreatedAtId<MyItemApiController>`, which is a helper method to create a response with a `201 Created` status code and a `Location` header.
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md:88]:
```suggestion
Now we can add an action to create a new item. We use the `HttpPost` attribute to define the `HTTP` method and route for the action.
Here we can see some validation logic. If the value does not start with "New", we return a `BadRequest` response with an error message. This highlights why we use the `OperationStatusResult` method. We can return a detailed response.

We also use `CreatedAtId<MyItemApiController>`, a helper method to create a response with a `201 Created` status code and a `Location` header.
```

File: 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+    return CreatedAtId<MyItemApiController>(
+        ctrl => nameof(ctrl.GetItem),
+        newItem.Id
+    );
+}
+```
+
+## Updating an item
+
+Now we can add an action to update an item. We use the `HttpPut` attribute to define the HTTP method and route for the action.
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md:128]:
```suggestion
Now we can add an action to update an item. We use the `HttpPut` attribute to define the `HTTP` method and route for the action.
```

File: 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+    }
+
+    item.Value = value;
+    return Ok();
+}
+```
+
+## Deleting an item
+
+Finally, we can add an action to delete an item. We use the `HttpDelete` attribute to define the HTTP method and route for the action.
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md:166]:
```suggestion
Finally, we can add an action to delete an item. We use the `HttpDelete` attribute to define the `HTTP` method and route for the action.
```

File: 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+            )
+        );
+    }
+
+    AllItems.Remove(item);
+    return Ok();
+}
+```
+## Full implementation
+Here is the controller with the full implementation.
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/create-your-own-api.md:191]:
```suggestion
Below is the full example of how the controller with the full implementation looks like.
```

**Review by jonat123 — APPROVED** [review]
@Zeegaan LGTM 😄

Before I merge it in please add the article to the summary.md file 😄

**Comment by Zeegaan** [comment]
@jonat123 done 🚀

**Pull request merged**